src/wallet/api/wallet2_api.h

```diff
@@ -74,44 +74,50 @@ enum NetworkType : uint8_t {
 */
 struct EnoteDetails
 {
-// from seraphis legacy enote types
     // Ko
-    rct::key onetime_address;
-    /// a
-    rct::xmr_amount amount;
-    /// C
-    rct::key amount_commitment;
-    /// enc(x)
-    rct::key encoded_amount_blinding_factor;
-    /// enc(a)
-    rct::key encoded_amount;
-    /// view_tag
-    crypto::view_tag view_tag;
-
-    // TODO : figure out if we need other members from transfer_details too, these are the ones that are not part of `TransactionInfo`, but my first impression is we could have an overlap so we can find TransactionInfo from EnoteDetails and vice versa.
-// from transfer_details:
-    cryptonote::transaction_prefix m_tx;
-    // index in m_tx.vout
-    uint64_t m_internal_output_index;
-    uint64_t m_global_output_index;
+    std::string m_onetime_address;
+    // view_tag
+    std::string m_view_tag;
+    // QUESTION : I'd argue block height for enote is redundant, you can get this information from TransactionHistory. Or any different opinions?
+//    std::uint64_t m_block_height;
+    // relative index in tx
+    std::uint64_t m_internal_output_index;
+    // absolute index from `cryptonote::COMMAND_RPC_GET_TRANSACTIONS::entry.output_indices`
+    std::uint64_t m_global_output_index;
+    // is spent
     bool m_spent;
+    // is frozen
     bool m_frozen;
-    uint64_t m_spent_height;
-    crypto::key_image m_key_image;
-    rct::key m_mask;
+    // blockchain height, set if spent
+    std::uint64_t m_spent_height;
+    // key image
+    std::string m_key_image;
+    // QUESTION : Is this comment correct?
+    // x, blinding factor in amount commitment C = x G + a H
+    std::string m_mask;
+    // a
+    std::uint64_t m_amount;
+    // QUESTION : Should we change m_rct to this enum to be prepared for the future? We could also add CryptoNote as cn before rct if that would make sense!?
```

Comment:
I vote for an enum, with now two values *cn* and *rct* to replace the boolean, and probably soon *fcmp* as a third value.

Over the years I developed a habit of quickly checking almost every boolean I see in an API how probable it seems to run into a problem with it after some expansions in the future.